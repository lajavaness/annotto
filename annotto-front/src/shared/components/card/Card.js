import PropTypes from 'prop-types'
import React from 'react'

import * as Styled from './__styles__/Card.styles'

const Card = ({ title, isContentFullSize, extra, children, ...otherProps }) => (
	<Styled.Root
		title={title}
		data-testid={'__card__'}
		bordered={false}
		$isContentFullSize={isContentFullSize}
		extra={extra}
		{...otherProps}
	>
		{children}
	</Styled.Root>
)

export default Card

Card.propTypes = {
	/** Defines the title of the card. */
	title: PropTypes.string,
	/** Defines the content to render in the right of the title of the card. */
	extra: PropTypes.node,
	/** Defines if the content should have padding. */
	isContentFullSize: PropTypes.bool,
	/** Defines the title of the card. */
	children: PropTypes.node,
}

Card.defaultProps = {
	title: null,
	extra: null,
	children: null,
	isContentFullSize: false,
}
