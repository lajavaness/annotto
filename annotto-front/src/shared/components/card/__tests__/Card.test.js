import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import React from 'react'

import theme from '__theme__'

import Card from 'shared/components/card/Card'

const defaultProps = {}
const getInstance = (props = {}) => (
	<ThemeProvider theme={theme}>
		<Card {...defaultProps} {...props} />
	</ThemeProvider>
)

describe('Card', () => {
	it('matches snapshot ', () => {
		const title = 'Foo'
		const { asFragment } = render(getInstance({ title }))
		expect(asFragment()).toMatchSnapshot()
	})
})
