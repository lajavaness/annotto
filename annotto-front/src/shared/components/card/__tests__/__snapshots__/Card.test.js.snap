// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="ant-card sc-bdfBQB gbIbhr"
    data-testid="__card__"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    />
  </div>
</DocumentFragment>
`;
